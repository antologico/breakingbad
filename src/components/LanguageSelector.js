import React from 'react'
import PropTypes from 'prop-types'
import { useDispatch } from 'react-redux'
import { setLanguage } from '../actions/language'
import messages from '../messages'
import { Select } from '../ui'
import { getLanguage } from '../selectors'

const LanguageSelector = ({ useDispatch, getLanguage, setLanguage }) => {
    const dispatch = useDispatch()
    const locale = getLanguage()
    return <Select data-test='language-selector' 
        value={locale}
        onChange={(e) => dispatch(setLanguage(e.target.value))}>
        {Object.keys(messages).map(key => 
            <option key={key} value={key}>
                {messages[key].name}
            </option>)
        }
    </Select>
}

LanguageSelector.propTypes = {
    getLanguage: PropTypes.func.isRequired,
    setLanguage: PropTypes.func.isRequired,
    useDispatch: PropTypes.func.isRequired,
}

LanguageSelector.defaultProps = {
    useDispatch,
    getLanguage,
    setLanguage,
}


export default LanguageSelector
