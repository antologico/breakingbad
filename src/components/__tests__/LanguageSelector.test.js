import React from 'react'
import { mount, create } from '../../tests/utils'
import LanguageSelector from '../LanguageSelector'

describe('LanguageSelector component', () => {
  test('renders properly', () => {
    const component = create(
        <LanguageSelector
          useDispatch={() => (fn) => fn}
          getLanguage={() => 'es'}
          setLanguage={() => {}}
        />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('calls setLanguage when changes', () => {
    const setLanguage = jest.fn()
    const component = mount(
        <LanguageSelector
        useDispatch={() => (fn) => fn}
        getLanguage={() => 'es'}
        setLanguage={setLanguage}
      />
    )
    component.update()
    const select = component.find('select').at(0)
    expect(component.find('select').exists()).toBe(true)
    select.simulate('change', { target: { value: 'en'} })
    expect(setLanguage).toBeCalledWith('en')
  })
})