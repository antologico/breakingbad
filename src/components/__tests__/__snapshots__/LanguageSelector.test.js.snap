// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector component renders properly 1`] = `
<select
  className="sc-jqUVSM kWFKVL"
  data-test="language-selector"
  onChange={[Function]}
  value="es"
>
  <option
    value="es"
  >
    Espa√±ol
  </option>
  <option
    value="en"
  >
    English
  </option>
</select>
`;
